### PR TITLE
Fix warp scan test

### PR DESCRIPTION
**Pull request by senior-zero**
Warp scan is not expected to be used with non-power-of-two logic warp sizes when there's more than one logic warp per hardware warp. Our test happened to be relying on that. This PR fixes warp scan test not to launch more threads than warp reduce expects.

### CODE REVIEW

File: test/test_warp_scan.cu [hunk 1]
```diff
@@ -399,7 +401,10 @@ void Test(
     InitialValueT   initial_value)
 {
     enum {
-        TOTAL_ITEMS = LOGICAL_WARP_THREADS * NUM_WARPS,
+        IS_ARCH_WARP = (LOGICAL_WARP_THREADS == CUB_WARP_THREADS(0)),
```
Comment (kaatish) [inline test/test_warp_scan.cu:404]: What is CUB_WARP_THREADS supposed to do?

File: test/test_warp_scan.cu [hunk 2]
```diff
@@ -458,7 +464,7 @@ void Test(
     fflush(stdout);
 
     // Run aggregate/prefix kernel
-    WarpScanKernel<LOGICAL_WARP_THREADS, TEST_MODE><<<1, TOTAL_ITEMS>>>(
+    WarpScanKernel<LOGICAL_WARP_THREADS, TEST_MODE><<<1, LOGICAL_WARP_THREADS * TOTAL_WARPS>>>(
```
Comment (kaatish) [inline test/test_warp_scan.cu:467]: Why was this change made? Isn't `TOTAL_ITEMS == LOGICAL_WARP_THREADS * TOTAL_WARPS`

File: test/test_warp_scan.cu (hunk 1 above)
Comment (senior-zero) [reply test/test_warp_scan.cu:404]: It represents hardware warp size throughout the project.

File: test/test_warp_scan.cu (hunk 2 above)
Comment (senior-zero) [reply test/test_warp_scan.cu:467]: It is, but items being processed doesn't immediately represent the number of threads you want. Now the code is written in terms of threads, not items.

**Review by kaatish — APPROVED** [review]

**Pull request merged**